request_handlers_test.cc

```diff
@@ -23,77 +24,84 @@ class RequestHandlersTest : public ::testing::Test {
 };
 
 TEST_F(RequestHandlersTest, EmptyStringTest) {
-  char request_buffer[MAX_REQ_SIZE];
+  char request_buffer[] = "";
+  const size_t request_buffer_size = 0;
 
-  // Customize message in request buffer
-  strcpy(request_buffer, "");
+  // The request_buffer is normally filled by a socket-read
+  // The response_buffer is allocated by the request-handler
+  char* response_buffer = nullptr;
+  size_t response_buffer_size = 0;
 
-  // Handle echo response in external handler
-  char response_buffer[MAX_REQ_SIZE];
-  size_t response_size = 0;
+  char expected_response_buffer[] =
+    "HTTP/1.1 200 OK\r\nContent-Type: text/plain\r\n\r\n";
+  const size_t expected_response_buffer_size = header_size + request_buffer_size;
```

Comment:
I don't currently see a way to do so, but it'd be nice if we were able to avoid repeatedly declaring expected sizes. Unfortunately, that's something that's only set/known after we make the call to the request-handler.